api/versions.go

```diff
@@ -326,12 +327,37 @@ func (self *version2_0) HandleRequest(requestType string, request []string, m ma
 		query := info.ContainerInfoRequest{
 			NumStats: sr.Count,
 		}
-		cont, err := m.GetContainerInfo(name, &query)
-		if err != nil {
-			return fmt.Errorf("failed to get container %q: %v", name, err)
+		if sr.IdType == "name" {
+			cont, err := m.GetContainerInfo(name, &query)
+			if err != nil {
+				return fmt.Errorf("failed to get container %q: %v", name, err)
+			}
+			contStats := convertStats(cont)
+			return writeResult(contStats, w)
 		}
-		contStats := convertStats(cont)
-		return writeResult(contStats, w)
+		if sr.IdType == "docker" {
+			contStats := make(map[string][]v2.ContainerStats, 0)
+			if name == "/" {
+				// special case: get all docker containers.
```

Comment:
What I mean is

/stats/?type=docker&recursive=true

Gives you all containers


/stats/?type=docker

Fails with: no Docker container specified.

It should still work with custom cgroups since it'd be like today in systemd systems.